src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,213 @@
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            var logonAccount = command.GetAutoLogonUserName();
+
+            string domainName;
+            string userName;
+
+            GetAccountSegments(logonAccount, out domainName, out userName);
+
+            if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+            {
+                logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+            }
+
+            Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+            string logonPassword = string.Empty;
+            var windowsServiceHelper = HostContext.GetService<INativeWindowsServiceHelper>();                
+            while (true)
+            {
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeed");
+                    break;
+                }
+                
+                if (command.Unattended)
+                {
+                    throw new SecurityException(StringUtil.Loc("InvalidWindowsCredential"));
+                }
+                    
+                Trace.Info("Invalid credential entered");
+                _terminal.WriteLine(StringUtil.Loc("InvalidWindowsCredential"));
+            }
+
+            bool isCurrentUserSameAsAutoLogonUser = windowsServiceHelper.IsTheSameUserLoggedIn(domainName, userName);                
+            var securityIdForTheUser = windowsServiceHelper.GetSecurityIdForTheUser(userName);
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            WindowsRegistryHelper regHelper = isCurrentUserSameAsAutoLogonUser 
+                                                ? new WindowsRegistryHelper(regManager) 
+                                                : new WindowsRegistryHelper(regManager, securityIdForTheUser);
+            
+            if(!isCurrentUserSameAsAutoLogonUser && !regHelper.ValidateIfRegistryExistsForTheUser(securityIdForTheUser))
+            {
+                Trace.Error(String.Format($"The autologon user '{logonAccount}' doesnt have a user profile on the machine. Please login once and reconfigure the agent agian"));
+                throw new InvalidOperationException("No user profile exists for the AutoLogon user");
+            }
+
+            DisplayUITestingRelatedWarningsIfAny(regHelper);
+            UpdateRegistrySettingsforUITesting(regHelper, userName, domainName, logonPassword);
+        }
+
+        private void DisplayUITestingRelatedWarningsIfAny(WindowsRegistryHelper regHelper)
+        {
+            var warningReasons = new List<string>();
```

Comment:
I have this message for the user when there is any warning, otherwise no message. _'WARNING: Following settings on the machine may affect the UI test runs on the machine.'_